compiler/z/codegen/OMRTreeEvaluator.cpp

```diff
@@ -10563,69 +10564,53 @@ OMR::Z::TreeEvaluator::arraycmpHelper(TR::Node *node,
       // Threshold for when to generate a preemptive CLC to avoid pulling in 2 cache lines if the arrays differ towards the beginning
       uint32_t earlyCLCThreshold = 32;
 
-      isEndLabelNeeded = loopIters || needResultReg || length > (earlyCLCEnabled ? earlyCLCThreshold : TR_MAX_CLC_SIZE);
+      uint8_t numDeps = 0;
+
+      if (baseSource1Ref->getBaseRegister())
+            numDeps++;
 
-      if (length <= (earlyCLCEnabled ? earlyCLCThreshold : TR_MAX_CLC_SIZE) && (isFoldedIf || !needResultReg))
+      if (baseSource2Ref->getBaseRegister() && (baseSource2Ref->getBaseRegister() != baseSource1Ref->getBaseRegister()))
```

Comment:
I think you can request a dependency with room to add baseSource2Ref's base register, irrespective of adding that to dep or not.

Reply:
Thanks. It looked strange to me as well but decided to leave it there since the original code was like that. I fix it.